### PR TITLE
FMN-539 Google Chat Bot code

### PR DESCRIPTION
**Pull request by CodeWarrior34**

### CODE REVIEW

File: src/main/resources/application.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+foreman:
+  baseUrl: https://dashboard.foreman.mn
+  apiUrl: https://api.foreman.mn
+
+google:
+  chatScope: https://www.googleapis.com/auth/chat.bot
+  bucket: dev-william-foreman-g-chat-bot
+  credentialOauthScope: https://www.googleapis.com/auth/cloud-platform
+  #resourceStream: C:\Users\WilliamHoward\Desktop\service-acct.json
```
Comment (delawr0190) [inline src/main/resources/application.yaml]: Remove

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    initialDelay: 60000
+    fixedDelay: 60000
+
+foreman:
+  baseUrl: https://dashboard.foreman.mn
+  apiUrl: https://api.foreman.mn
+
+google:
+  chatScope: https://www.googleapis.com/auth/chat.bot
+  bucket: dev-william-foreman-g-chat-bot
```
Comment (delawr0190) [inline src/main/resources/application.yaml]: In production app.yml, you need an environment variable:

```
google.bucket: "foreman-google-chat-bot"
```

File: src/main/java/mn/foreman/googelchatbot/session/SessionRepository.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import java.io.IOException;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+
+/** The storage repository for the {@link Session}. */
+public interface SessionRepository {
+
+    /** Deletes a blob from the bucket. */
+    void delete(String spaceId);
```
Comment (delawr0190) [inline src/main/java/mn/foreman/googelchatbot/session/SessionRepository.java:16]: Missing doc on the param.

```
@param
```

File: src/main/java/mn/foreman/googelchatbot/session/SessionRepository.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    /* Finds all sessions */
```
Comment (delawr0190) [inline src/main/java/mn/foreman/googelchatbot/session/SessionRepository.java]: Missing doc on the:

```
@return
```

File: src/main/java/mn/foreman/googelchatbot/notifications/NotificationsProcessor.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+import java.io.IOException;
+
+public interface NotificationsProcessor {
+
+    /**
+     * Obtains notifications for the provided session and notifies the chat, as
+     * necessary.
+     *
+     * @param session Provides the users credentials.
```
Comment (delawr0190) [inline src/main/java/mn/foreman/googelchatbot/notifications/NotificationsProcessor.java:14]: Missing

```
@param
```

On the second arg.

**Review by delawr0190 — CHANGES_REQUESTED** [review]

File: src/main/resources/application.yaml (hunk 2 above)
Comment (CodeWarrior34) [reply src/main/resources/application.yaml]: Ok. I was planning on letting the default named app.yaml be the production one.

**Review by delawr0190 — APPROVED** [review]

**Pull request merged**